# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsDragService.cpp
# Commit: ddd973acce13
# Full Hash: ddd973acce13685abfd86ba44e9094a7b26c78d0
# Author: stransky <stransky@redhat.com>
# Date: 2024-12-11 10:02:50
# Description:
#   Bug 1935861 [Linux] Don't check MIME type for mDragData r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231683
# ==============================================================================

diff -r 54110d753e29 -r ddd973acce13 widget/gtk/nsDragService.cpp
--- a/widget/gtk/nsDragService.cpp	Tue Dec 10 13:28:38 2024 +0000
+++ b/widget/gtk/nsDragService.cpp	Tue Dec 10 13:29:16 2024 +0000
@@ -489,17 +489,11 @@
 
 bool DragData::IsDataValid() const {
   if (mDragData) {
-    if (IsTextFlavor() || IsURIFlavor()) {
-      return mDragData.get() && mDragDataLen;
-    }
-    MOZ_DIAGNOSTIC_ASSERT(false, "DragData::IsDataValid(): Unknow MIME type.");
-    return false;
+    return mDragData.get() && mDragDataLen;
   } else if (mDragUris) {
     return !!(mDragUris.get()[0]);
-  } else if (mUris.Length()) {
+  } else {
     return mUris.Length();
-  } else {
-    return false;
   }
 }
 
